client/my-sites/themes/theme-preview.jsx

```diff
@@ -27,6 +27,8 @@ import { connectOptions } from './theme-options';
 
 const DEFAULT_VARIATION_SLUG = 'default';
 const isDefaultVariationSlug = ( slug ) => ! slug || slug === DEFAULT_VARIATION_SLUG;
+const isUrlWpcomApi = ( url ) =>
+	url.indexOf( 'public-api.wordpress.com/wpcom/v2/block-previews/site' ) >= 0;
```

Comment:
I guess we can simply check the origin is `public-api.wordpress.com`. Or we can use `isExternallyManagedTheme` to check whether it's 3rd party theme. WDYT?

Reply:
That's much better, thanks for the suggestion!